### PR TITLE
Update API doc

### PR DESCRIPTION
**Pull request by variadico**
This adds documentation to methods that were missing it. In addition, it also fixes some minor comment mismatches. I also did a little bit of tweaking to existing doc.

### CODE REVIEW

File: js.go [hunk 1]
```diff
@@ -1478,31 +1490,35 @@ func (p ReplayPolicy) MarshalJSON() ([]byte, error) {
...
-	AckAck      = []byte("+ACK")
-	AckNak      = []byte("-NAK")
-	AckProgress = []byte("+WPI")
-	AckNext     = []byte("+NXT")
-	AckTerm     = []byte("+TERM")
+	ackAck      = []byte("+ACK")
+	ackNak      = []byte("-NAK")
+	ackProgress = []byte("+WPI")
+	ackNext     = []byte("+NXT")
+	ackTerm     = []byte("+TERM")
```
Comment (variadico) [inline js.go:1497]: I wasn't totally sure about this change. I didn't see a place that required the public user to pass any of these variables to us, so I decided to hide them instead.

Comment (wallyqs) [reply js.go:1497]: sounds good to me to keep them private.

**Review by wallyqs — APPROVED** [review]
LGTM

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**